docs/content/docs/try-flink/write_flink_program_with_sql.md

```diff
@@ -0,0 +1,269 @@
+
+To find out what gaming streams started in the last 15 minutes, try this query:
+
+```sql
+SELECT game_name, 
+       started_at,
+       proctime, 
+       TIMESTAMPDIFF(MINUTE, started_at, proctime)
+FROM twitch_stream
+WHERE TIMESTAMPDIFF(MINUTE, started_at, proctime) < 15;
+```
+
+You should now see a new table with new datastream results.  
+
+
+Now try a [Top-N]({{< ref "docs/dev/table/sql/queries/topn" >}}) query to find the 10 most popular games.
+
+Top-N queries identify the N smallest or largest values (as ordered by some attribute of the table), and are useful when you need to identify the top (or bottom) N items in a stream. 
+
+```sql
+SELECT game_name, proctime, viewer_count
+FROM (
+    SELECT *, ROW_NUMBER() OVER (ORDER BY viewer_count DESC) AS ranking
+    FROM (
+        SELECT *
+        FROM (
+            SELECT 
+                *,
+                ROW_NUMBER() OVER (PARTITION BY game_name ORDER BY proctime DESC) AS ordering
+            FROM twitch_stream
+        ) WHERE ordering = 1
+    )
+) WHERE ranking <= 10;
+```
+
+Note that Flink uses the combination of an OVER window clause and a filter condition to express a Top-N query in order to filter through unbounded dataset. 
+
+You should now see a new dynamic table with the results from this query.  Notice how the top 10 games are constantly being revised as new data is processed.
+
+
+## Step 6: Aggregate the data and learn about windowing 
+
+Now try an aggregate function (`COUNT`, `SUM`, `AVG`, `MIN`, `MAX`) to find the most popular games based on location with the following query: 
+
+```sql
+SELECT location,
+       game_name,
+       COUNT(game_name) 
+FROM twitch_stream
+WHERE location IS NOT NULL
+GROUP BY location, game_name;
+```
+
+You should see a table result of continously changing data that tells you the most popular games for each country and the count for each game. 
+
+For the last example, let's try a very commonly used function in streaming SQL: computing windowed aggregations.
+
+Windows are the fundamental building blocks when it comes to processing infinite streams since they provide a way to split streams into finite chunks. 
+Flink SQL offers several [window functions]({{< ref "docs/dev/table/sql/queries/window-tvf" >}}) that you can use.  
+
+Let's explore how to apply one of these functions! To calculate an aggregation per game of the total viewer count for every 1 minute window, 
+you can use the [tumble]({{< ref "docs/dev/table/sql/queries/window-tvf#tumble" >}}) function.
+
+Tumbling windows can be thought of as mini-batches of aggregations over a non-overlapping window of time.
+
+```sql
+SELECT window_start,
+       window_end,
+       game_name,
+       SUM(viewer_count) AS TotalViewerCount
+  FROM TABLE(
+    TUMBLE(TABLE twitch_stream, DESCRIPTOR(proctime), INTERVAL '1' MINUTE))
+  GROUP BY window_start, window_end, game_name;
+```
+
+Fresh results will appear every minute, showing the sum of the viewer counts for each game.
```

Comment:
I found a minute to be rather a long time to wait. Maybe we should shorten the windows.